web/modules/custom/dbcdk_community/src/Profile/ProfileRepository.php

```diff
@@ -0,0 +1,357 @@
+<?php
+/**
+ * @file
+ * Class definition for ProfileRepository.
+ */
+
+namespace Drupal\dbcdk_community\Profile;
+
+
+use DateTime;
+use DBCDK\CommunityServices\Api\CommunityRoleApi;
+use DBCDK\CommunityServices\Api\ProfileApi;
+use DBCDK\CommunityServices\Api\QuarantineApi;
+use DBCDK\CommunityServices\Model\CommunityRole;
+use DBCDK\CommunityServices\Model\Profile as ModelProfile;
+use DBCDK\CommunityServices\Model\Quarantine;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+
+/**
+ * A repository for retrieving profiles with related objects.
+ *
+ * Related objects can be quarantines and community roles.
+ */
+class ProfileRepository {
+  use LoggerAwareTrait;
+
+  /**
+   * The API to use when retrieving profiles.
+   *
+   * @var \DBCDK\CommunityServices\Api\ProfileApi $profileApi
+   */
+  protected $profileApi;
+
+  /**
+   * The API to use when retrieving quarantines.
+   *
+   * @var \DBCDK\CommunityServices\Api\QuarantineApi $quarantineApi
+   */
+  protected $quarantineApi;
+
+  /**
+   * The API to use when retrieving community roles.
+   *
+   * @var \DBCDK\CommunityServices\Api\CommunityRoleApi
+   */
+  protected $communityRoleApi;
+
+  /**
+   * ProfileRepository constructor.
+   *
+   * @param \DBCDK\CommunityServices\Api\ProfileApi $profile_api
+   *   The profile api to use.
+   * @param \DBCDK\CommunityServices\Api\QuarantineApi $quarantine_api
+   *   The quarantine api to use.
+   * @param \DBCDK\CommunityServices\Api\CommunityRoleApi $community_role_api
+   *   The community role api to use.
+   */
+  public function __construct(ProfileApi $profile_api, QuarantineApi $quarantine_api, CommunityRoleApi $community_role_api) {
+    $this->profileApi = $profile_api;
+    $this->quarantineApi = $quarantine_api;
+    $this->communityRoleApi = $community_role_api;
+    $this->logger = new NullLogger();
```

Comment:
I do not think that makes much sense. The fact that it exists is defined when using the trait. I would not think about setting variables of a parent class either.

Would you prefer this alternate version?

```php
$this->setLogger(new NullLogger());
```